File: go/internal/models/hotelModel.go
package models

type Hotel struct {
	//gorm.Model
	ID          uint    `gorm:"primary_key; not null" json:"id"`
	Name        string  `gorm:"size:255; not null" json:"name"`
	Destination string  `gorm:"size:64; not null" json:"destination"`
	Price       float64 `gorm:"null" json:"price"`
	Image       string  `gorm:"size:255; not null" json:"image"`
	Status      int64   `gorm:"default:0;not null" json:"status"`
	Address     string  `gorm:"size:255;null;" json:"address"`
	Longitude   float64 `gorm:"default:0;not null" json:"longitude"`
	Latitude    float64 `gorm:"default:0;not null" json:"latitude"`
}

File: flutter_app/lib/models/hotel_model.dart
enum HotelStatus { ongoing, done }

class Hotel {
  int id;
  String name;
  String image;
  String destination;
  double price;
  HotelStatus status;
  LocationHotel? location;
  Hotel({
    required this.name,
    required this.destination,
    required this.image,
    required this.price,
    required this.id,
    this.location,
    this.status = HotelStatus.ongoing,
  });


  String toString() {
    return 'Hotel{id: $id, name: $name,destination: $destination, price: $price}';
  }

  Hotel.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        name = json['name'],
        destination = json['destination'],
        image = json['image'],
        price = json['price'].toDouble(),
        location = LocationHotel(
          address: json['address'],
          latitude: json['latitude'],
          longitude: json['longitude'],
  ),
  status =
  json['status'] == 0 ? HotelStatus.ongoing : HotelStatus.done;

  Map<String, dynamic> toJson() {
    Map<String, dynamic> value = {
      'id': id,
      'name': name,
      'image': image,
      'price': price,
      'address': location?.address,
      'longitude': location?.longitude,
      'latitude': location?.latitude,
      'status': status == HotelStatus.ongoing ? 0 : 1
    };
    value['id'] = id;
    return value;
  }
}

class LocationHotel {
  String? address;
  double? longitude;
  double? latitude;
  LocationHotel({
    this.address,
    this.longitude,
    this.latitude,
  });
}


